### PR TITLE
feature/jobs-screen-event

### PR DESCRIPTION
**Pull request by Belal-Alkhatib**

### CODE REVIEW

**Review by AmeerAmjed — APPROVED** [review]

File: data/local/src/main/java/com/cupcake/jobsfinder/local/daos/JobFinderDao.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import androidx.room.Delete
+import androidx.room.Insert
+import androidx.room.OnConflictStrategy
+import androidx.room.Query
+import com.cupcake.jobsfinder.local.entities.JobsEntity
+
+@Dao
+interface JobFinderDao {
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    fun insertJop(job: JobsEntity)
```
Comment (msalarab) [inline data/local/src/main/java/com/cupcake/jobsfinder/local/daos/JobFinderDao.kt]: Jop!!

File: data/remote/src/main/java/com/cupcake/remote/interceptor/AuthIntercepter.kt [hunk 2]
```diff
@@ -26,7 +26,7 @@ override fun intercept(chain: Interceptor.Chain): Response {
     private companion object {
         const val AUTHORIZATION = "Authorization"
         const val BEARER = "bearer"
-         val TOKEN = ""
+         val TOKEN = "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJlZTVhZTBkNS0xNzgzLTRhNWItYWVkMi0zYzcxOWQyY2FmMzQiLCJleHAiOjE2ODc0NTg3MjN9.KuUkrlsOBP75uQq5HDkd-7ZEQZ0_zleiPYOUdV3rPRk"
```
Comment (msalarab) [inline data/remote/src/main/java/com/cupcake/remote/interceptor/AuthIntercepter.kt]: remove it

File: domain/usecase/src/main/java/com/cupcake/usecase/SaveJobUseCase.kt [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import repo.JobFinderRepository
+import javax.inject.Inject
+
+class SaveJobUseCase @Inject constructor(
+    private val repository: JobFinderRepository,
+) {
+    suspend operator fun invoke(job: JobWithTitle): Boolean {
+        if(isAlreadyExist(job.id)){
+            deleteInsertedJob(job)
+            return false //that mean the saved job are deleted
```
Comment (msalarab) [inline domain/usecase/src/main/java/com/cupcake/usecase/SaveJobUseCase.kt:13]: remove this comment

File: presentation/ui/src/main/java/com/cupcake/ui/utill/BindingAdapter.kt [hunk 4]
```diff
@@ -90,4 +93,20 @@ fun showDrawableErrorImage(view: ImageView, errorType: BaseErrorUiState?){
...
    view.text = convert(time)
-}
\ No newline at end of file
+}
+
+@OptIn(InternalCoroutinesApi::class)
+@BindingAdapter("setIsSavedIcon")
+fun setIsSavedIcon(textView: TextView, isSaved: Boolean) {
+    val drawable: Drawable = if (isSaved) {
+        ContextCompat.getDrawable(textView.context, R.drawable.ic_saved)!!
```
Comment (msalarab) [inline presentation/ui/src/main/java/com/cupcake/ui/utill/BindingAdapter.kt]: remove !!

File: presentation/ui/src/main/res/layout/edit_job_post_bottom_sheets.xml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_marginHorizontal="@dimen/size16">
+
+            <com.google.android.material.bottomsheet.BottomSheetDragHandleView
+                android:id="@+id/drag_handle"
+                android:layout_width="@dimen/bottom_sheet_drag_handle_view_width"
+                android:layout_height="@dimen/bottom_sheet_drag_handle_view_height"
+                android:layout_marginVertical="@dimen/size16"
+                android:layout_marginTop="@dimen/size16"
```
Comment (msalarab) [inline presentation/ui/src/main/res/layout/edit_job_post_bottom_sheets.xml]: remove it and using spacing_medium

File: presentation/ui/src/main/res/layout/edit_job_post_bottom_sheets.xml [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toTopOf="parent" />
+
+            <TextView
+                android:id="@+id/textViewSave"
+                style="@style/TextStyle.body"
+                android:layout_width="match_parent"
+                android:layout_height="@dimen/button_height"
+                android:layout_marginVertical="@dimen/size20"
```
Comment (msalarab) [inline presentation/ui/src/main/res/layout/edit_job_post_bottom_sheets.xml]: the same

File: presentation/ui/src/main/res/layout/fragment_jobs.xml [hunk 7]
```diff
@@ -1,81 +1,109 @@
...
+            <com.google.android.material.floatingactionbutton.FloatingActionButton
+                android:id="@+id/fab"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:src="@drawable/ic_add"
+                android:backgroundTint="#4B8CED"
+                app:layout_constraintBottom_toBottomOf="parent"
+                app:layout_constraintEnd_toEndOf="parent"
+                android:layout_margin="25dp"
+                app:tint="#FFFFFF"
```
Comment (msalarab) [inline presentation/ui/src/main/res/layout/fragment_jobs.xml]: use color from values

File: presentation/ui/src/main/res/layout/fragment_jobs.xml [hunk 8]
```diff
@@ -1,81 +1,109 @@
...
+                app:layout_constraintTop_toBottomOf="@id/error"
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintStart_toStartOf="parent"
+                />
+            <com.google.android.material.floatingactionbutton.FloatingActionButton
+                android:id="@+id/fab"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:src="@drawable/ic_add"
+                android:backgroundTint="#4B8CED"
```
Comment (msalarab) [inline presentation/ui/src/main/res/layout/fragment_jobs.xml]: added to color file

File: presentation/ui/src/main/res/layout/fragment_jobs.xml [hunk 9]
```diff
@@ -1,81 +1,109 @@
...
+                android:text="@string/retry"
+                app:showWhenIf="@{viewModel.state.error != null}"
+                android:onClick="@{() -> viewModel.onRetryClicked()}"
+                app:circularflow_defaultRadius="@dimen/spacing_large"
+                app:layout_constraintTop_toBottomOf="@id/error"
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintStart_toStartOf="parent"
+                />
+            <com.google.android.material.floatingactionbutton.FloatingActionButton
+                android:id="@+id/fab"
```
Comment (msalarab) [inline presentation/ui/src/main/res/layout/fragment_jobs.xml]: rename id

File: presentation/ui/src/main/res/layout/fragment_jobs.xml [hunk 10]
```diff
@@ -1,81 +1,109 @@
...
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:src="@drawable/ic_add"
+                android:backgroundTint="#4B8CED"
+                app:layout_constraintBottom_toBottomOf="parent"
+                app:layout_constraintEnd_toEndOf="parent"
+                android:layout_margin="25dp"
```
Comment (msalarab) [inline presentation/ui/src/main/res/layout/fragment_jobs.xml]: what!

File: presentation/ui/src/main/res/values-night/colors.xml [hunk 11]
```diff
@@ -7,6 +7,8 @@
     <color name="accent">#14FFFFFF</color>
     <color name="cards">#222222</color>
     <color name="blue_300">#303843</color>
+    <color name="white">#212121</color>
```
Comment (msalarab) [inline presentation/ui/src/main/res/values-night/colors.xml:10]: !!!

File: presentation/ui/src/main/res/values/dimens.xml [hunk 12]
```diff
@@ -19,6 +19,8 @@
     <dimen name="text_size_body">14sp</dimen>
     <dimen name="text_size_caption">12sp</dimen>
     <dimen name="text_size_small">10sp</dimen>
+    <dimen name="text_size_very_small">8sp</dimen>
```
Comment (msalarab) [inline presentation/ui/src/main/res/values/dimens.xml]: remove it

File: presentation/ui/src/main/res/values/strings.xml [hunk 13]
```diff
@@ -116,6 +116,9 @@
     <string name="required_skills">Required Skills</string>
     <string name="popular_jobs">Popular Jobs</string>
     <string name="retry">Retry</string>
-
-
+    <string name="save">Save</string>
+    <string name="share">Share</string>
+    <string name="sharePost">Hello bro,Your Post is very awesome , select your App to share \n%1$s . </string>
+    <string name="saved_successfully">Saved Successfully</string>
+    <string name="saved_canceld">Saved Canceled</string>
```
Comment (msalarab) [inline presentation/ui/src/main/res/values/strings.xml]: e

File: presentation/viewmodels/src/main/java/com/cupcake/viewmodels/jobs/BottomSheetViewModel.kt [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+                    )
+                }
+            }
+
+        }
+    }
+
+
+
+    private fun onJobSaveSuccess(isSaved: Boolean) {
```
Comment (msalarab) [inline presentation/viewmodels/src/main/java/com/cupcake/viewmodels/jobs/BottomSheetViewModel.kt]: ??!

File: presentation/viewmodels/src/main/java/com/cupcake/viewmodels/jobs/JobsEvent.kt [hunk 15]
```diff
@@ -1,6 +1,15 @@
 package com.cupcake.viewmodels.jobs
 
+
 sealed class JobsEvent {
     data class JobCardClick(val id: String) : JobsEvent()
-    data class JobChipClick(val id: String) : JobsEvent()
+
+    object OnFloatingActionClickListener : JobsEvent()
+
+    data class OnImageViewMoreClickListener(val model:JobUiState): JobsEvent()
```
Comment (mustafa-sab3alarab) [inline presentation/viewmodels/src/main/java/com/cupcake/viewmodels/jobs/JobsEvent.kt:9]: OnMoreOptionClickListener

**Review by msalarab — CHANGES_REQUESTED** [review]
change requests

**Pull request merged**